src/Carbon/Carbon.php

```diff
@@ -234,6 +258,11 @@ public function __construct($time = null, $tz = null)
             $time = $testInstance->toDateTimeString();
         }
 
+        if ($time === null || $time === 'now') {
+            $dateTime = new DateTime('now', static::safeCreateDateTimeZone($tz));
+            $time = $dateTime->format('Y-m-d H:i:s').'.'.substr(microtime(), 2, 6);
```

Comment:
Some time passes between the `DateTime` call on line 262 and the `microtime` call on 263.  It may be fractions of a second or whole seconds, depending on the speed of the system that executes the code.  It would be better to get a `DateTime` objects that includes fractional seconds, then use the `u` format to include it in the string:

```php
$dateTime = \DateTime::createFromFormat('U.u', strval(microtime(true)));
$time = $dateTime->format('Y-m-d H:i:s.u'); // include μs
```

The `u` format gives the fraction as microseconds (six decimal places).  To get only milliseconds (three decimal places), use `substr` to strip off the last three characters:

```php
$time = substr($dateTime->format('Y-m-d H:i:s.u'), 0, -3); // truncate μs to ms
```